docs/yeti-branching-strategy.md

```diff
@@ -0,0 +1,35 @@
+# Yeti Branching Strategy
+
+This document describes Yeti's preferred strategy for organizing branches with Git.  
+
+The exact pattern will vary from project to project, but this is what we do when the choice is up to us.  
+
+This approach also assumes that we have both a deployed staging and production environment.  
+
+## What's the Approach?
+
+Our typical approach is essentially [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow), which was popularized [around 2010](https://nvie.com/posts/a-successful-git-branching-model/).
+
+We have two main branches: `main` and `develop`.  
+
+The `develop` branch is configured to automatically deploy to the staging environment. This branch is the most up-to-date version of the codebase, containing the most recent development commits.  
+
+The `main` branch is configured to automatically deploy to the production environment, and should only contain merge commits from `develop` and any urgent production `hotfix` commits.  
+
+## How Do We Create Feature Branches?
+
+We create feature branches off of the `develop` branch.  
+
+Feature branches should have a descriptive name. We tend to follow the [conventional commits specification](https://www.conventionalcommits.org/en/v1.0.0/), which works quite well for branch names as well.  
+
+For example, if we needed to add a widget generation service to a project we would create a new feature branch off of the `develop` branch:
+
+```bash
+git checkout -b feat/add-widget-generation-service
+```
+
+When the feature is ready for review we'd open a pull request against the `develop` branch. Once the feature is approved it is **squashed and merged** into `develop` and automatically deployed to staging.  
+
+## How Do We Deploy to Production?
+
+Deploying to production should be fairly simple. All we need to do is merge `develop` into `main` and the production deployment should start automatically.
```

Comment:
Is it worth documenting our merge strategy here? Merge commit, rebase, etc. We've also had some accidental squash and merges in the past from `develop` to `main`.

Reply:
This is a good callout, thanks. I'll add to this tomorrow. 👍